examples/dmd/parametric_dmdc_heat_conduction.cpp

```diff
@@ -0,0 +1,1070 @@
+                fout.close();
+            }
+
+            const CAROM::Matrix* control_mat = createControlMatrix(d_controls);
+
+            std::string full_file_name;
+
+            full_file_name = io_dir + "/" + to_string(alpha) + "_" + to_string(
+                                 kappa) + "_" + to_string(
+                                 amp_in) + "_" + to_string(amp_out) + "_control";
+            control_mat->write(full_file_name);
+        }
+
+
+        if (online)
+        {
+            if (myid == 0)
+            {
+                std::cout << "Creating DMD using the rdim of the offline phase" << std::endl;
+            }
+
+            std::fstream fin(io_dir + "/parameters.txt", std::ios_base::in);
+            double curr_param;
+            std::vector<std::string> dmdc_paths;
+            std::vector<CAROM::Matrix*> controls;
+            std::vector<CAROM::Vector*> param_vectors;
+
+            while (fin >> curr_param)
+            {
+                double curr_alpha = curr_param;
+                fin >> curr_param;
+                double curr_kappa = curr_param;
+                fin >> curr_param;
+                double curr_amp_in = curr_param;
+                fin >> curr_param;
+                double curr_amp_out = curr_param;
+
+                dmdc_paths.push_back(io_dir + "/" + to_string(curr_alpha) + "_" + to_string(
+                                         curr_kappa) + "_" + to_string(curr_amp_in) + "_" + to_string(curr_amp_out) );
+
+                CAROM::Matrix* curr_control = new CAROM::Matrix();
+                curr_control->read(io_dir + "/" + to_string(curr_alpha) + "_" + to_string(
+                                       curr_kappa) + "_" + to_string(curr_amp_in) + "_" + to_string(
+                                       curr_amp_out) + "_control");
+                controls.push_back(curr_control);
+
+                CAROM::Vector* param_vector = new CAROM::Vector(4, false);
+                param_vector->item(0) = curr_alpha;
+                param_vector->item(1) = curr_kappa;
+                param_vector->item(2) = curr_amp_in;
+                param_vector->item(3) = curr_amp_out;
+                param_vectors.push_back(param_vector);
+            }
+            fin.close();
+
+            CAROM::Vector* desired_param = new CAROM::Vector(4, false);
+            desired_param->item(0) = alpha;
+            desired_param->item(1) = kappa;
+            desired_param->item(2) = amp_in;
+            desired_param->item(3) = amp_out;
+
+            dmd_training_timer.Start();
+
+
+            CAROM::Matrix* controls_interpolated = new CAROM::Matrix();
+
+            CAROM::getParametricDMDc(dmd_u, param_vectors, dmdc_paths, controls,
+                                     controls_interpolated, desired_param, "G", "LS", closest_rbf_val);
+
+            dmd_u->project(init,controls_interpolated);
+
+
+            dmd_training_timer.Stop();
+            delete desired_param;
```

Comment:
There are a few other variables that are allocated but not freed: `controls_interpolated`, and the pointers inside of `controls` and `param_vectors`.

Reply:
For some reason I'm getting a segmentation fault when trying to delete pointers inside`controls` here. Specifically where deleting `controls[2]` and `controls[3]`. I need to figure out why.

Reply:
Good catch! Changing `MatrixInterpolator` based on your suggestion fixes the segmentation fault and still gives the same results as before. Thanks!